amethyst_ui/src/shaders/vertex.glsl

```diff
@@ -0,0 +1,27 @@
+// TODO: Needs documentation.
+
+#version 150 core
+
+layout (std140) uniform VertexArgs {
+    uniform vec4 proj_vec;
+    uniform vec2 coord;
+    uniform vec2 dimension;
+};
+
+in vec3 position;
+in vec2 tex_coord;
+
+out VertexData {
+  vec4 position;
+  vec2 tex_coord;
+} vertex;
+
+void main() {
+    vertex.position = vec4(position, 1);
+    vertex.position *= vec4(dimension, 1, 1);
+    vertex.position += vec4(coord, 0, 0);
+    vertex.position *= proj_vec;
+    vertex.position += vec4(-1, 1, 0, 0);
```

Comment:
Why aren't you just using a matrix?

Reply:
This is more efficient, less multiplying by 1 and adding 0.

Reply:
I'm not sure if we can measure GPU performance in terms of how often a multiplication is happening, but rather how typical and thus optimized it is for a certain situation. So I don't know what's more efficient here, it's just unconventional.

Reply:
Vec4 operations are gonna be faster than s matrix for the most part

Reply:
So I did this because @omni-viral and  @Rhuagh told me the projection matrix was excessive.

Reply:
You can even reduce it further using vec2 and swizzle operators if you want.